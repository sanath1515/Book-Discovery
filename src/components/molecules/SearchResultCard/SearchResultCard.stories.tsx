import React from 'react';
import SearchResultCard from './SearchResultCard';

export default {
    title: 'molecules/SearchResultCard',
    component: SearchResultCard
};

const Default = (args) => <SearchResultCard title="Nature Chemistry"
authorName="J D Lee" field="Chemistry"
image="https://cdn.zeplin.io/5f03041d8c8fc11335546618/assets/7AE68B73-7512-4AED-BE9F-881F1BB4B173.png" id={''}/>;

export const Primary = Default.bind({});

