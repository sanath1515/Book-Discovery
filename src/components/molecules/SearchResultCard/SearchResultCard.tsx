import React from 'react';
import { makeStyles } from '@material-ui/core/styles';
import Card from '@material-ui/core/Card';
import CardActionArea from '@material-ui/core/CardActionArea';
import CardContent from '@material-ui/core/CardContent';
import CardMedia from '@material-ui/core/CardMedia';
import { CardActions, Typography } from '@material-ui/core';
import { COLORS } from '../../../theme/constants';
import theme from '../../../theme/theme';
import { useDispatch } from "react-redux";
import { useHistory } from "react-router";

const useStyles = makeStyles({
    root: {
        width: "358px",
        height: "90px",
        padding: theme.spacing(1, 1, 1, 8),
        borderRadius: "4px",
        boxShadow: "none",
        border:"1px solid",
        borderColor:COLORS.GREY_300,
    },
    image: {
        width: "60px",
        height: "75px",
        margin: theme.spacing(2,2),
        objectFit: "contain"
    },
    title: {
        width: "263px",
        height: "22px",
        margin: theme.spacing(1),
        color: COLORS.CHARCOAL_GREY_100,
        textAlign:"left"
    },
    author: {
        width: "14px",
        height: "18px",
        color: COLORS.ICON_500,
        textTransform: "none"
    },
    authorName: {
        width: "50px",
        height: "18px",
        color: COLORS.CHARCOAL_GREY,
        textTransform: "none",
        textAlign: "left"
    },
    field: {
        width: "32px",
        height: "18px",
        textTransform: "none",
        color: COLORS.ICON_500,
    },
    fieldName: {
        width: "55px",
        height: "18px",
        textTransform: "none",
        Color: COLORS.CHARCOAL_GREY_100
    },
    actions: {
        margin: theme.spacing(0),
        padding: theme.spacing(1)
    },
    actionArea: {
        display: "flex",
        width: "auto",
    },
    content:{
        padding: theme.spacing(0)
    }
});

export type SearchResultCardProps={
    title: string;
    authorName: string;
    field: string;
    image: string;
    id: string;
    handleClose:()=>void;
}

const SearchResultCard: React.FC<SearchResultCardProps> = ({
    title,
    field,
    authorName,
    image,
    id,
    handleClose
}) => {
    const dispatch = useDispatch();
    const history = useHistory();
    const handleOnClick=(bookId:string)=>{
        dispatch({ type: "SET_SINGLE_VIEW_ID", payload: bookId });
        history.push("/bookdetail");
        handleClose();
    }
    const classes = useStyles();
    return (
        <Card className={classes.root}>
            <CardActionArea className={classes.actionArea}>
                <CardMedia className={classes.image}
                    component="img"
                    image={image} onClick={()=>handleOnClick(id)} data-testid="card"/>
                <CardContent className={classes.content}>
                    <Typography className={classes.title} variant="body2">{title}</Typography>
                    <CardActions className={classes.actions}>
                        <Typography className={classes.author} variant="overline">by</Typography>
                        <Typography className={classes.authorName} variant="caption">{authorName}</Typography>
                    </CardActions>
                    <CardActions className={classes.actions}>
                        <Typography className={classes.field} variant="overline">Field :</Typography>
                        <Typography className={classes.fieldName} variant="overline">{field}</Typography>
                    </CardActions>
                </CardContent>
            </CardActionArea>
        </Card>
    );
}

export default SearchResultCard;