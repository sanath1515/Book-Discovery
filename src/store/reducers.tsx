export type bookState = {
    searchText:string,
    category:string,
    singleViewId:string
}

export const initialBookState = {
    searchText:"",
    category:"",
    singleViewId:"1"
}

type Action = {type:string,payload:string}

export const booksReducer = (state:bookState = initialBookState,action:Action)=>
{
    switch(action.type)
    {
        case "SET_CATEGORY":{
            return {...state,category:action.payload};
        }
        case "SET_SEARCH_TEXT":{
            return {...state,searchText:action.payload};
        }
        case "SET_SINGLE_VIEW_ID":{
            return {...state,singleViewId:action.payload};
        }
        default:
            return state;
    }
}