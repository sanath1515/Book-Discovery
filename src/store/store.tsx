import { createStore } from 'redux';
import { booksReducer } from './reducers';

const store = createStore(booksReducer);

export default store;