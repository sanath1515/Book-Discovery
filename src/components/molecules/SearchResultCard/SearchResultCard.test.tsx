import { render, screen, cleanup, fireEvent } from "@testing-library/react";
import SearchResultCard from "./SearchResultCard";
import { Provider } from "react-redux";
import { BrowserRouter as Router } from "react-router-dom";
import store from "../../../store/store";

afterEach(cleanup);

describe("Search Result Card Testing", () => {
  it("Matches Snapshot", () => {
    const { container } = render(
      <Router>
        <Provider store={store}>
          <SearchResultCard
            title="Nature Chemistry"
            field="Chemistry"
            authorName="J D Lee"
            image="https://cdn.zeplin.io/5f03041d8c8fc11335546618/assets/920F4926-3BD5-47FB-A32A-32C03DE1BB71.png"
            id={""} handleClose={()=>{}}          />
        </Provider>
      </Router>
    );
    expect(container).toMatchSnapshot();
  });
  it("Checking The title", () => {
    render(
      <Router>
        <Provider store={store}>
          <SearchResultCard
            title="Nature Chemistry"
            field="Chemistry"
            authorName="J D Lee"
            image="https://cdn.zeplin.io/5f03041d8c8fc11335546618/assets/920F4926-3BD5-47FB-A32A-32C03DE1BB71.png"
            id={""} handleClose={()=>{}}     />
        </Provider>
      </Router>
    );

    expect(screen.getByText("Nature Chemistry")).toBeInTheDocument();
    expect(screen.getByText("J D Lee")).toBeInTheDocument();
    expect(screen.getByText("Chemistry")).toBeInTheDocument();
  });
  it("Should click correctly", () => {
    const { getByTestId } = render(
      <Router>
        <Provider store={store}>
        <SearchResultCard
            title="Nature Chemistry"
            field="Chemistry"
            authorName="J D Lee"
            image="https://cdn.zeplin.io/5f03041d8c8fc11335546618/assets/920F4926-3BD5-47FB-A32A-32C03DE1BB71.png"
            id={""} handleClose={()=>{}}      />
        </Provider>
      </Router>
    );
    fireEvent.click(getByTestId("card"));
    expect(fireEvent.click(getByTestId("card"))).toBeTruthy();
  });
});
