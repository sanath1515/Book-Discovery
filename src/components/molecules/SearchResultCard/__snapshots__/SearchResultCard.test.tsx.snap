// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search Result Card Testing Matches Snapshot 1`] = `
<div>
  <div
    class="MuiPaper-root MuiCard-root makeStyles-root-1 MuiPaper-elevation1 MuiPaper-rounded"
  >
    <button
      class="MuiButtonBase-root MuiCardActionArea-root makeStyles-actionArea-9"
      tabindex="0"
      type="button"
    >
      <img
        class="MuiCardMedia-root makeStyles-image-2 MuiCardMedia-media MuiCardMedia-img"
        data-testid="card"
        src="https://cdn.zeplin.io/5f03041d8c8fc11335546618/assets/920F4926-3BD5-47FB-A32A-32C03DE1BB71.png"
      />
      <div
        class="MuiCardContent-root makeStyles-content-10"
      >
        <p
          class="MuiTypography-root makeStyles-title-3 MuiTypography-body2"
        >
          Nature Chemistry
        </p>
        <div
          class="MuiCardActions-root makeStyles-actions-8 MuiCardActions-spacing"
        >
          <span
            class="MuiTypography-root makeStyles-author-4 MuiTypography-overline"
          >
            by
          </span>
          <span
            class="MuiTypography-root makeStyles-authorName-5 MuiTypography-caption"
          >
            J D Lee
          </span>
        </div>
        <div
          class="MuiCardActions-root makeStyles-actions-8 MuiCardActions-spacing"
        >
          <span
            class="MuiTypography-root makeStyles-field-6 MuiTypography-overline"
          >
            Field :
          </span>
          <span
            class="MuiTypography-root makeStyles-fieldName-7 MuiTypography-overline"
          >
            Chemistry
          </span>
        </div>
      </div>
      <span
        class="MuiCardActionArea-focusHighlight"
      />
      <span
        class="MuiTouchRipple-root"
      />
    </button>
  </div>
</div>
`;
